src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -84,4 +85,37 @@ private function findAdderMethod($object, $property)
             return $method;
         }
     }
+
+    /**
+     * @param string $plural
+     *
+     * @return string
+     */
+    private function singularize($plural)
+    {
+        if (null === $this->singularizeClass || null === $this->singularizeMethod) {
+            // default to false so we only try this once
+            $this->singularizeClass = false;
```

Comment:
As you have only 1 instance of `ArrayAdd` used for all objects, if you leave two objects `Dummy` and `AnotherDummy`, when calling `findAdderMethod()` on a `Dummy` object, you will have `ArrayAdd#singularizeClass === 'Dummy'`. This is fine if the next object called is a `Dummy` object, but when the next object will be `AnotherDummy`, `findAdderMethod()` will return a false result.